mypy/subtypes.py

```diff
@@ -1055,6 +1055,20 @@ def visit_type_type(self, left: TypeType) -> bool:
                 #       of x is Type[int]. It's unclear what's the right way to address this.
                 return True
             item = left.item
+            if (
+                right.type.is_protocol
+                and (len(right.type.protocol_members) == 1)
+                and (right.type.protocol_members[0] == "__hash__")
```

Comment:
This would behave incorrectly in an exotic case where the protocol contains a member named `__hash__` that doesn't have the standard signature. Maybe that's OK though as it's better than the current state.

Reply:
The same holds for the potential subclass.  I added additional checks for both of them.